modules/sr/robot/vision/vectors.py

```diff
@@ -156,6 +156,12 @@ def angle_between(vec_a: Vector, vec_b: Vector) -> float:
     dp = dot_product(vec_a, vec_b)
     mod_ab = vec_a.magnitude() * vec_b.magnitude()
     cos_theta = dp / mod_ab
+
+    if abs(cos_theta) > 1:
+        # Round small floating point rounding errors to avoid a math domain
+        # error from math.acos, without masking genuine errors.
+        cos_theta = round(cos_theta, 15)
```

Comment:
Any reason why 15dp? Magic numbers scare me!

Reply:
Because there are 18 significant figures of precision in a `float` and we should (in theory at least) be looking at a number with magnitude less than 1.

I picked a number which was likely to be sufficiently clear of any rounding errors while still preserving plenty of precision.